OpenRA.Mods.Common/UpdateRules/Rules/20231010/AddSpawnActorPowerBlockedCursor.cs

```diff
@@ -0,0 +1,36 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright (c) The OpenRA Developers and Contributors
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System.Collections.Generic;
+
+namespace OpenRA.Mods.Common.UpdateRules.Rules
+{
+	public class AddSpawnActorPowerBlockedCursor : UpdateRule
+	{
+		public override string Name => "Add SpawnActorPower BlockedCursor.";
+
+		public override string Description =>
+			"SpawnActorPower BlockedCursor changed its default value.";
+
+		public override IEnumerable<string> UpdateActorNode(ModData modData, MiniYamlNodeBuilder actorNode)
+		{
+			var spawnActorPower = actorNode.LastChildMatching("SpawnActorPower", false);
+			if (spawnActorPower == null)
+				yield break;
+
+			if (spawnActorPower.LastChildMatching("BlockedCursor", false) != null)
+				yield break;
+
+			var blockedCursorNode = new MiniYamlNodeBuilder("BlockedCursor", new MiniYamlBuilder("move-blocked"));
+			spawnActorPower.AddNode(blockedCursorNode);
```

Comment:
```suggestion
			foreach (var spawnActorPower in actorNode.ChildrenMatching("SpawnActorPower"))
			{
				var cursor = spawnActorPower.LastChildMatching("BlockedCursor");
				if (cursor != null && !cursor.IsRemoval())
					yield break;

				var blockedCursorNode = new MiniYamlNodeBuilder("BlockedCursor", new MiniYamlBuilder("move-blocked"));
				spawnActorPower.AddNode(blockedCursorNode);
			}
```

this should be a foreach. Also doesn't GrantExternalConditionPower also need the update rule?

Reply:
Why does `GrantExternalConditionPower` need an update rule?